File: algebra/triangle_area/cpp/HeronsFormula.cpp
#include<iostream>
#include<cmath>

using namespace std;

float areaFromSides(float a, float b, float c)
{
	float s = (a+b+c)/2;
	return sqrt(s*(s-a)*(s-b)*(s-c));
}


int main()
{
	float a, b, c;
	cout<<endl<<"Enter the three sides of the triangle :"<<endl;
	cin>>a>>b>>c;
	cout<<"The area of the triangle is "<< areaFromSides(a,b,c)<<" units"<<endl;
	return 0;
}

File: algebra/triangle_area/rust/herons_formula.rs
fn area_from_sides(a: f64, b: f64, c: f64) -> f64 {
    let s = (a + b + c) / 2.0;
    (s * (s - a) * (s - b) * (s - c)).sqrt()
}

fn main() {
    println!("{}", area_from_sides(2.0, 3.0, 4.0))
}
